types/slonik/index.d.ts

```diff
@@ -281,20 +304,24 @@ export interface SqlTaggedTemplateType {
     ) => ArraySqlTokenType;
     identifier: (
         names: string[]
-    ) => IdentifierTokenType;
+    ) => IdentifierSqlTokenType;
     json: (
         value: SerializableValueType
     ) => JsonSqlTokenType;
+    join: (
+      members: ReadonlyArray<ValueExpressionType>,
+      glue: SqlTokenType,
+    ) => ListSqlTokenType;
     raw: (
         rawSql: string,
-        values?: PrimitiveValueExpressionType[]
-    ) => RawSqlTokenType;
+        values?: ReadonlyArray<PrimitiveValueExpressionType>
+    ) => SqlTokenType;
     unnest: (
-        // Value might be PrimitiveValueExpressionType[],
+        // Value might be ReadonlyArray<ReadonlyArray<PrimitiveValueExpressionType>>,
         // or it can be infinitely nested array, e.g.
         // https://github.com/gajus/slonik/issues/44
-        tuples: any[][],
-        columnTypes: string[]
+        tuples: ReadonlyArray<ReadonlyArray<any>>,
+        columnTypes: ReadonlyArray<string>
```

Comment:
What's the benefit of requiring `ReadonlyArray<string>`? It's fine if the function is always used inline, but slightly more inconvenient to work with otherwise, since now I can't do:

```typescript
const tuples = [['a', 'b']]
const columnTypes = ['foo', 'bar']
...
slonik.unnest(tuples, columnTypes)
```

Reply:
I added the `ReadonlyArray` type in line with the [original Flow types](https://github.com/gajus/slonik/blob/master/src/types.js#L349). To my understanding there are no restrictions to the array passed in as parameter, but the `ReadonlyArray` would simply imply that the function itself is not performing any mutations on the provided array.